<launch>
	<!-- base params -->
	<arg name="mav_addr" default="udpin:0.0.0.0:14552" />
	<arg name="arm_topic" default="controller/isArmed" />
	<arg name="pcl_topic" default="controller/usePCL" />

	<!-- Goal topic (XYZ) and point to focus on (currently only using xy position) -->
	<arg name="focus_topic" default="controller/focusPoint" />
	<arg name="goal_topic" default="controller/goalPoint" />
	<arg name="goal_path_topic" default="controller/goalPath" />

	<!-- Segment size (maximum step) in metres and goal threshold -->
	<arg name="segment_size_topic" default="controller/segmentSize" />
	<arg name="goal_threshold_topic" default="controller/goalThreshold" />

	<!-- Frames to monitor current location of the ROV -->
	<arg name="world_frame" default="world" />
	<arg name="rov_frame" default="base_link" />

	<arg name="log_output" default="screen" />
	<arg name="respawn_control" default="false" />

	<node pkg="controls" type="controller.py" name="controller" required="$(eval not respawn_control)"
	    clear_params="true" output="$(arg log_output)" respawn="$(arg respawn_control)">
		<param name="mav_addr" value="$(arg mav_addr)" />
		<param name="arm_topic" value="$(arg arm_topic)" />
		<param name="pcl_topic" value="$(arg pcl_topic)" />
		<param name="focus_topic" value="$(arg focus_topic)" />
		<param name="goal_topic" value="$(arg goal_topic)" />
		<param name="goal_path_topic" value="$(arg goal_path_topic)" />
		<param name="segment_size_topic" value="$(arg segment_size_topic)" />
		<param name="goal_threshold_topic" value="$(arg goal_threshold_topic)" />
		<param name="world_frame" value="$(arg world_frame)" />
		<param name="rov_frame" value="$(arg rov_frame)" />
	</node>
</launch>